sklearn/utils/_mocking.py

```diff
@@ -64,6 +67,11 @@ class CheckingClassifier(ClassifierMixin, BaseEstimator):
     check_y_params, check_X_params : dict, default=None
         The optional parameters to pass to `check_X` and `check_y`.
 
+    methods_to_check : "all" or list of str, default="all"
```

Comment:
Do we need a test where `methods_to_check` is more than one element? i.e. `methods_to_check=["predict", "score"]` to check that `predict`, and `score` fails and  `decision_function` passes?

Reply:
I can add this test.